File: tmp-migrate.js
const fs = require('fs');
const path = require('path');

const SOURCE_DIR = 'admin';
const TARGET_DIR = 'admin-vite/src';

const DIRS_TO_COPY = ['components', 'lib', 'store', 'types'];

function copyDirIterative(src, dest) {
    if (!fs.existsSync(dest)) fs.mkdirSync(dest, { recursive: true });
    const entries = fs.readdirSync(src, { withFileTypes: true });
    for (let entry of entries) {
        const srcPath = path.join(src, entry.name);
        const destPath = path.join(dest, entry.name);
        if (entry.isDirectory()) {
            copyDirIterative(srcPath, destPath);
        } else {
            fs.copyFileSync(srcPath, destPath);
        }
    }
}

for (let d of DIRS_TO_COPY) {
    const src = path.join(SOURCE_DIR, d);
    const dst = path.join(TARGET_DIR, d);
    if (fs.existsSync(src)) {
        if (fs.existsSync(dst)) fs.rmSync(dst, { recursive: true, force: true });
        copyDirIterative(src, dst);
    }
}

const pagesDir = path.join(TARGET_DIR, 'pages');
if (!fs.existsSync(pagesDir)) fs.mkdirSync(pagesDir, { recursive: true });

function processContent(content) {
    content = content.replace(/['"]use client['"];?\n?/g, '');
    content = content.replace(/import Image from ['"]next\/image['"];?\n?/g, '');
    content = content.replace(/<Image([^>]+)\/>/g, '<img$1/>');
    content = content.replace(/import {?\\s*Link\\s*}? from ['"]next\/link['"];?/g, 'import { Link } from "react-router-dom";');
    content = content.replace(/import\s+{([^}]+)}\s+from\s+['"]next\/navigation['"];?/g, 'import { useNavigate, useLocation } from "react-router-dom";');
    content = content.replace(/useRouter\(\)/g, 'useNavigate()');
    content = content.replace(/const\s+pathname\s*=\s*usePathname\(\);?/g, 'const { pathname } = useLocation();');
    content = content.replace(/router\.push/g, 'navigate');
    content = content.replace(/router\.replace/g, 'navigate');
    return content;
}

function processDir(directory) {
    if (!fs.existsSync(directory)) return;
    const entries = fs.readdirSync(directory, { withFileTypes: true });
    for (let entry of entries) {
        const fullPath = path.join(directory, entry.name);
        if (entry.isDirectory()) {
            processDir(fullPath);
        } else if (/\.(ts|tsx|js|jsx)$/.test(entry.name)) {
            let content = fs.readFileSync(fullPath, 'utf8');
            content = processContent(content);
            fs.writeFileSync(fullPath, content, 'utf8');
        }
    }
}

for (let d of DIRS_TO_COPY) {
    processDir(path.join(TARGET_DIR, d));
}

const mapping = [
    ['page.tsx', 'LoginPage.tsx'],
    ['dashboard/page.tsx', 'DashboardPage.tsx'],
    ['dashboard/layout.tsx', 'DashboardLayout.tsx'],
    ['dashboard/categories/page.tsx', 'CategoriesPage.tsx'],
    ['dashboard/orders/page.tsx', 'OrdersPage.tsx'],
    ['dashboard/specialists/page.tsx', 'SpecialistsPage.tsx'],
    ['dashboard/users/page.tsx', 'UsersPage.tsx'],
    ['providers.tsx', 'Providers.tsx']
];

for (let [srcFile, dstName] of mapping) {
    const fullSrc = path.join(SOURCE_DIR, 'app', srcFile);
    const fullDst = path.join(pagesDir, dstName);
    if (fs.existsSync(fullSrc)) {
        let content = fs.readFileSync(fullSrc, 'utf8');
        content = processContent(content);
        fs.writeFileSync(fullDst, content, 'utf8');
    }
}

console.log("Migration complete!");

File: tmp-migrate.py
import os
import shutil
import re

SOURCE_DIR = 'admin'
TARGET_DIR = 'admin-vite/src'

DIRS_TO_COPY = ['components', 'lib', 'store', 'types']

# Copy top-level folders
for d in DIRS_TO_COPY:
    src = os.path.join(SOURCE_DIR, d)
    dst = os.path.join(TARGET_DIR, d)
    if os.path.isdir(src):
        if os.path.exists(dst):
            shutil.rmtree(dst)
        shutil.copytree(src, dst)

# Create pages dir
os.makedirs(os.path.join(TARGET_DIR, 'pages'), exist_ok=True)

# Helper for string replacements
def process_file_content(content):
    # Convert 'use client'
    content = re.sub(r'[\'"]use client[\'"];?\n?', '', content)
    
    # Replace next/image
    content = re.sub(r'import Image from [\'"]next/image[\'"];?\n?', '', content)
    content = re.sub(r'<Image([^>]+)/>', r'<img\1 />', content) # Naive Image replacement
    
    # Replace next/link
    content = re.sub(r'import Link from [\'"]next/link[\'"];?', 'import { Link } from "react-router-dom";', content)
    
    # Replace next/navigation
    content = re.sub(r'import\s+{([^}]+)}\s+from\s+[\'"]next/navigation[\'"];?', 'import { useNavigate, useLocation } from "react-router-dom";', content)
    content = re.sub(r'useRouter\(\)', 'useNavigate()', content)
    content = re.sub(r'const\s+pathname\s*=\s*usePathname\(\);?', 'const { pathname } = useLocation();', content)
    
    # Replace router.push, router.replace
    content = re.sub(r'router\.push', 'navigate', content)
    content = re.sub(r'router\.replace', 'navigate', content) # Note: technically navigate('/path', {replace: true}) but naive is fine for now
    
    # Replace @/ imports
    content = re.sub(r'from [\'"]@/', 'from \'@/', content)
    
    return content

# Recursively process copied files
def process_dir(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.ts', '.tsx', '.js', '.jsx')):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                new_content = process_file_content(content)
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(new_content)

for d in DIRS_TO_COPY:
    dst = os.path.join(TARGET_DIR, d)
    if os.path.exists(dst):
        process_dir(dst)

# Extract pages from app/
app_dir = os.path.join(SOURCE_DIR, 'app')
pages_dir = os.path.join(TARGET_DIR, 'pages')

mapping = [
    ('page.tsx', 'LoginPage.tsx'),
    ('dashboard/page.tsx', 'DashboardPage.tsx'),
    ('dashboard/layout.tsx', 'DashboardLayout.tsx'),
    ('dashboard/categories/page.tsx', 'CategoriesPage.tsx'),
    ('dashboard/orders/page.tsx', 'OrdersPage.tsx'),
    ('dashboard/specialists/page.tsx', 'SpecialistsPage.tsx'),
    ('dashboard/users/page.tsx', 'UsersPage.tsx')
]

for src_file, dst_name in mapping:
    full_src = os.path.join(app_dir, src_file)
    full_dst = os.path.join(pages_dir, dst_name)
    if os.path.exists(full_src):
        with open(full_src, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Replace default export with named export logic or just keeping it
        content = process_file_content(content)
        
        with open(full_dst, 'w', encoding='utf-8') as f:
            f.write(content)

print("Migration script completed base copy!")
